### PR TITLE
Secrets C/R/U/D

### PR DESCRIPTION
**Pull request by petey**
* https fixes for UI
* new route for pipeline secret settings `/pipelines/:id/secrets`
* Secret model should handle crud operations as requested by user

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -45,7 +46,8 @@ module.exports = (environment) => {
 
   if (environment === 'production') {
     // special config for production
-    ENV.APP.SDAPI_HOSTNAME = 'http://api.screwdriver.cd';
+    ENV.APP.SDAPI_HOSTNAME = 'https://api.screwdriver.cd';
+    ENV.APP.SDUI_HOSTNAME = 'https://cd.screwdriver.cd';
```
Comment (tkyi) [inline config/environment.js]: 👍

**Comment by stjohnjohnson** [comment]
I'm not sure how I feel about the name `environment` since this is _only_ about secrets.  All non-secret environment variables are controlled via the configuration.

**Comment by jer** [comment]
👍

**Comment by stjohnjohnson** [comment]
Excellent update! 👍 

catbox

**Pull request merged**